docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+this subclass-of-`Command` object is returned to `WerkIt#startContinuousUserPrompt()`.
+
+> A detailed implementation of the parsing and creation of subclass-of-`Command` object process can be found in
+'[Parsing User Input and Getting the Right Command](#parsing-user-input-and-getting-the-right-command)'.
+
+Next, `WerkIt#startContinuousUserPrompt()` calls on the `execute()` method of the subclass-of-`Command` object to
+perform the user's requested action. If the execution goes smoothly, this completes the user's inputted command.
+This process is repeated until the user enters `exit`, which will terminate the loop, call `UI#printGoodbye()` to
+print a goodbye message to the user, before handing control back to `Main#main` to end the program.
+
+#### Design Considerations
+* `WerkIt#startContinuousUserPompt()` has a boolean flag `isFirstPrompt`. This flag allows WerkIt to
+print a different prompt each time the application starts up, before defaulting to a different prompt message
+for subsequent prompts.
+   * When the user starts the application, `isFirstPrompt` is set to `true` and thus, the prompt will be
+  ```
+  ----------------------------------------------------------------------
+  Now then, what can I do for you today?
+  (Need help? Type 'help' for a guide!)
+  ----------------------------------------------------------------------
+  >
+  ```
+  * Subsequent prompts in that app's session will be
+  ```
+  ----------------------------------------------------------------------
+  What's next?
+  ----------------------------------------------------------------------
+  >
+  ```
+
+---
+
+### Parsing User Input and Getting the Right Command
+**_TODO_**: Explain how the app parses user input and determines which `Command` subclass object to instantiate.
+
+---
+
+### Exercise
+#### List Exercise
+
+
+If the user's command type is to list the exercises available, i.e. `exercise /list`, the
+`Parser#parseUserInput(String userInput)` method will parse the 'exercise' base word and proceed to create exercise related
+command using `Parser#createExerciseCommand(String userInput)` method. This method will further evaluate the
+exercise action, in this case, `/list` and call the constructor of `ExerciseCommand` class by passing relevant parameters related to the
+ExerciseCommand constructor. If the exercise action is null or incorrect, an InvalidCommandException will be thrown. Once the exercise command is created,
+this exercise command is executed via the `ExerciseCommand#execute()` method. As it is executed, the method will check the
+type of action to be executed, in this case, list. It will then list the exercises available for selection from the exerciseList using the `ExerciseList#printExerciseList()`.
+
+The following sequence diagram illustrates how the `exercise /list` command works in greater detail:
+
+![List Exercise Sequence Diagram](uml/sequenceDiagrams/images/viewExercise.png)
```

Comment:
Based on what I understand of the code, `parseUserInput()` constructs and returns a `Command` object (an `ExerciseCommand` in this case). In that case, I am wondering why the invocation of `ExerciseCommand`'s constructor is not shown here, and a random unnamed `Command` object is returned in step 5 instead. It's not immediately obvious from the diagram that this `Command` object is (I am assuming) the same one as the `newCommand:ExerciseCommand` object in the sequence diagram. I've indicated the relevant portions in red.

Also, I think it's possible to remove the boxes from the bottom of the sequence diagram (marked in blue) if you're using PlantUML. It might be good to do that to adhere to the module's format. (This goes for all sequence diagrams in this document, too.)

![image](https://user-images.githubusercontent.com/26963639/160962597-02de4f17-37e3-4a13-ab82-53c4124b9b22.png)